enclave-tls/src/include/enclave-tls/tls_wrapper.h

```diff
@@ -0,0 +1,111 @@
+/* *INDENT-OFF* */
+#ifndef _ENCLAVE_TLS_WRAPPER_H
+#define _ENCLAVE_TLS_WRAPPER_H
+/* *INDENT-ON* */
+
+#include <stdint.h>
+#include <stddef.h>
+
+#include <enclave-tls/err.h>
+#include <enclave-tls/api.h>
+
+#define TLS_WRAPPER_TYPE_MAX                32
+#define TLS_WRAPPER_API_VERSION_1           1
+#define TLS_WRAPPER_API_VERSION_DEFAULT     TLS_WRAPPER_API_VERSION_1
+
+#define TLS_WRAPPER_OPTS_FLAGS_SGX_ENCLAVE  1
+
+#define MAX_ENCLAVE_TLS_NAME                32
+#define ENCLAVE_QUOTE_TYPE_MAX              32
+
+typedef struct {
+	struct tls_wrapper_opts_t *opts;
+	void *tls_private;
+	unsigned long conf_flags;
+	enclave_tls_log_level_t log_level;
+	void *handle;
+	int fd;
+} tls_wrapper_ctx_t;
+
+typedef struct {
+	const char *organization;
+	const char *organization_unit;
+	const char *common_name;
+} cert_subject_t;
+
+typedef struct {
+	uint8_t ias_report[2 * 1024];
+	uint32_t ias_report_len;
+	uint8_t ias_sign_ca_cert[2 * 1024];
+	uint32_t ias_sign_ca_cert_len;
+	uint8_t ias_sign_cert[2 * 1024];
+	uint32_t ias_sign_cert_len;
+	uint8_t ias_report_signature[2 * 1024];
+	uint32_t ias_report_signature_len;
+} attestation_verification_report_t;
+
+typedef struct {
+	uint8_t quote[2048];
+	uint32_t quote_len;
+	/* Certificiate in PEM format. */
+	uint8_t pck_crt[2048];
+	uint32_t pck_crt_len;
+	/* Certificate chain in PEM format. */
+	uint8_t pck_sign_chain[4096];
+	uint32_t pck_sign_chain_len;
+	/* JSON data as published by https://api.portal.trustedservices.intel.com/documentation#pcs-tcb-info */
+	uint8_t tcb_info[4096];
+	uint32_t tcb_info_len;
+	/* Certificate chain in PEM format. */
+	uint8_t tcb_sign_chain[4096];
+	uint32_t tcb_sign_chain_len;
+	/* JSON data, e.g., as obtained from https://api.portal.trustedservices.intel.com/documentation#pcs-qe-identity */
+	uint8_t qe_identity[1024];
+	uint32_t qe_identity_len;
+	/* PEM-encoded CRL as published by https://certificates.trustedservices.intel.com/IntelSGXRootCA.crl */
+	uint8_t root_ca_crl[1024];
+	uint32_t root_ca_crl_len;
+	/* PEM-encoded certificate revocation list as published by https://api.portal.trustedservices.intel.com/documentation#pcs-revocation */
+	uint8_t pck_crl[1024];
+	uint32_t pck_crl_len;
+} ecdsa_attestation_evidence_t;
+
+typedef struct {
+	const char type[ENCLAVE_QUOTE_TYPE_MAX];
```

Comment:
ENCLAVE_QUOTE_TYPE_MAX is used to define the max supported types of quote not the length of type name.